prow/cmd/gerrit/main.go

```diff
@@ -68,8 +80,15 @@ func (o *options) Validate() error {
 		return errors.New("--last-sync-fallback must be set")
 	}
 
-	if strings.HasPrefix(o.lastSyncFallback, "gs://") && o.gcsCredentialsFile == "" {
-		logrus.WithField("last-sync-fallback", o.lastSyncFallback).Warn("--gcs-credentials-file unset, will try and access with a default service account")
+	if o.gcsCredentialsFile != "" {
+		logrus.Warning("-gcs-credentials-file is deprecated and will be removed in September 2020. Use -blob-storage-credentials-file instead.")
+	}
+	if o.gcsCredentialsFile != "" && o.blobStorageCredentialsFile != "" {
+		return fmt.Errorf("please set either -blob-storage-credentials-file or -gcs-credentials-file but not both")
```

Comment:
This is not true. We should not require a credentials file to be present on GCP (and prow.k8s.io no longer uses them). Best practice is to use workload identity

Reply:
So I guess in case no credentialsFile is specified `storage.NewClient` detects the workload identity and no error is returned from `storage.NewClient` here: https://github.com/kubernetes/test-infra/blob/master/pkg/io/opener.go#L60-L68

Reply:
Removed the flag validation